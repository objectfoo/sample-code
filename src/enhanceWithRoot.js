
import React, { Component } from 'react';
import injectSheet, {ThemeProvider} from 'react-jss';
import theme from './theme.json';

let AppWrapper = props => props.children;
const styles = theme => {
  // console.log(theme);
  return {
    '@global': {
      'body': {
        fontFamily: theme.typeography.fontFamily,
        fontSize: theme.typeography.fontSize,
        lineHeight: theme.typeography.lineHeight
      },
      'body *': {
        'box-sizing': 'border-box'
      },
      'html, body': {
        margin: 0,
        padding: 0
      },
      'html, body, #root': {
        height: '100%'
      }
    }
  };
};

AppWrapper = injectSheet(styles)(AppWrapper);

function enhanceWithRoot (BaseComponent) {
  class enhanceWithRoot extends Component {
    componentDidMount () {
      // Remove the server-side injected CSS.
      const jssStyles = document.querySelector('#jss-server-side');
      if (jssStyles && jssStyles.parentNode) {
        jssStyles.parentNode.removeChild(jssStyles);
      }
    }

    render () {
      return (
        <ThemeProvider theme={theme}>
          <AppWrapper>
            <BaseComponent {...this.props} />
          </AppWrapper>
        </ThemeProvider>
      );
    }
  }

  return enhanceWithRoot;
}

export default enhanceWithRoot;
