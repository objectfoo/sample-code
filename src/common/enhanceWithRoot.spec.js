import React from 'react';
import enhanceWithRoot from './enhanceWithRoot';
import {shallow} from 'enzyme';
import {Dummy} from '../test/fixtures/Dummy';

describe('enhanceWithRoot', () => {
  it('should enhance a base component', () => {
    const Component = enhanceWithRoot(Dummy);
    const wrapper = shallow(<Component />);

    expect(wrapper).toMatchSnapshot();
  });
});
