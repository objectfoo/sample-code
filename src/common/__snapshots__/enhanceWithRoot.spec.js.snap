// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enhanceWithRoot should enhance a base component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Jss(AppWrapper)>
        <Dummy />
</Jss(AppWrapper)>,
      "theme": Object {
        "palette": Object {
          "error": Object {
            "500": "#f44336",
          },
          "primary": Object {
            "500": "#2196F3",
          },
          "secondary": Object {
            "500": "#FF9800",
          },
          "text": Object {
            "disabled": "rgba(0, 0, 0, 0.38)",
            "primary": "rgba(0, 0, 0, 0.87)",
            "secondary": "rgba(0, 0, 0, 0.54)",
          },
        },
        "spacing": Object {
          "unit": 8,
        },
        "typeography": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": 15,
          "fontWeightLight": 300,
          "fontWeightMedium": 500,
          "fontWeightRegular": 400,
          "lineHeight": 1.6,
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Dummy />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Jss(AppWrapper)>
          <Dummy />
</Jss(AppWrapper)>,
        "theme": Object {
          "palette": Object {
            "error": Object {
              "500": "#f44336",
            },
            "primary": Object {
              "500": "#2196F3",
            },
            "secondary": Object {
              "500": "#FF9800",
            },
            "text": Object {
              "disabled": "rgba(0, 0, 0, 0.38)",
              "primary": "rgba(0, 0, 0, 0.87)",
              "secondary": "rgba(0, 0, 0, 0.54)",
            },
          },
          "spacing": Object {
            "unit": 8,
          },
          "typeography": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": 15,
            "fontWeightLight": 300,
            "fontWeightMedium": 500,
            "fontWeightRegular": 400,
            "lineHeight": 1.6,
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Dummy />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
